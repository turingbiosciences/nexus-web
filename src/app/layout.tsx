import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "./globals.css";
import { ReactQueryProvider } from "@/components/providers/react-query-provider";
import { AuthProvider } from "@/components/auth/auth-provider";
import { GlobalAuthProvider } from "@/components/providers/global-auth-provider";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Turing Biosciences",
  description: "Secure file upload and management platform",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <body className={inter.className}>
        <ReactQueryProvider>
          <AuthProvider>
            <GlobalAuthProvider>{children}</GlobalAuthProvider>
          </AuthProvider>
        </ReactQueryProvider>
      </body>
    </html>
  );
}
