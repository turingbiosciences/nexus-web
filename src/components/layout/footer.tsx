"use client";

export function Footer() {
  const currentYear = new Date().getFullYear();

  return (
    <footer className="bg-white border-t mt-auto">
      <div className="container-page py-6">
        <p className="text-center text-sm text-gray-600">
          &copy; {currentYear} Turing Biosciences. All rights reserved.
        </p>
      </div>
    </footer>
  );
}
