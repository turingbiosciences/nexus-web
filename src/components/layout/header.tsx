"use client";

import { AuthButton } from "@/components/auth/auth-button";
import { Logo } from "@/components/ui/logo";

export function Header() {
  return (
    <header className="bg-white shadow-sm border-b">
      <div className="container-page">
        <div className="flex justify-between items-center py-4">
          <div className="flex items-center">
            <Logo />
          </div>
          <AuthButton />
        </div>
      </div>
    </header>
  );
}
