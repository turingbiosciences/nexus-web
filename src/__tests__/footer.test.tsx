import { render, screen } from "@testing-library/react";
import { Footer } from "@/components/layout/footer";
import React from "react";

const THIS_YEAR = new Date().getFullYear();

describe("Footer", () => {
  it("renders current year and company", () => {
    render(<Footer />);
    expect(
      screen.getByText(new RegExp(`${THIS_YEAR} Turing Biosciences`, "i"))
    ).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
