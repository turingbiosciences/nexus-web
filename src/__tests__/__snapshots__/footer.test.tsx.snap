// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<footer
  class="bg-white border-t mt-auto"
>
  <div
    class="container-page py-6"
  >
    <p
      class="text-center text-sm text-gray-600"
    >
      Â© 
      2025
       Turing Biosciences. All rights reserved.
    </p>
  </div>
</footer>
`;
